app/helpers/laboratory_query_helper.rb

```diff
@@ -5,4 +5,87 @@ module LaboratoryQueryHelper
   def laboratories_by_patient_ids(patient_ids)
     Laboratory.where(patient_id: patient_ids).order(:patient_id)
   end
+
+  # Validates the request params for the index action
+  def validate_laboratory_query(params)
+    permitted_params = params.permit(:entries, :page, :search, :order, :direction)
+
+    patient_id = params.require(:patient_id)&.to_i
+    search_text = permitted_params[:search]
+    sort_order = permitted_params[:order]
+    sort_direction = permitted_params[:direction]
+    # Assume default values if pagination data is not explicitly specified
+    entries = permitted_params[:entries]&.to_i || 10
+    page = permitted_params[:page]&.to_i || 0
+
+    if sort_direction.present? && %w[asc desc].exclude?(sort_direction)
+      error_message = "Unable to sort column in specified direction in request: '#{sort_direction}'"
+      raise StandardError, error_message
+    end
+
+    if sort_order.present? && %w[id lab_type specimen_collection report result].exclude?(sort_order)
+      error_message = "Unable to sort by specified column in request: '#{sort_order}'"
+      raise StandardError, error_message
+    end
+
+    # Either both the sort order and direction are provided, or neither is provided
+    if (sort_direction.blank? && sort_order.present?) || (sort_direction.present? && sort_order.blank?)
+      error_message = 'Must have both a sort column and direction specified or neither specified. Requested column to sort: '\
+                      "'#{sort_order}'', with specified direction: '#{sort_direction}'"
+      raise StandardError, error_message
+    end
+
+    raise InvalidQueryError.new(:entries, entries) if entries.negative?
+    raise InvalidQueryError.new(:page, page) if page.negative?
+
+    {
+      patient_id: patient_id,
+      search_text: search_text,
+      sort_order: sort_order,
+      sort_direction: sort_direction,
+      entries: entries,
+      page: page
+    }
+  end
+
+  # Queries laboratories by ID or type based on search text.
+  def search(laboratories, search)
+    return laboratories if search.blank?
+
+    laboratories.where('id like ?', "#{search}%").or(
+      laboratories.where('lab_type like ?', "%#{search&.downcase}%").or(
+        laboratories.where('result like ?', "%#{search&.downcase}%")
+      )
+    )
+  end
+
```

Comment:
It looks like in the previous Lab Results table, the user could search by year (found via UI testing). Just confirming that we no longer want this search. (It looks like the other tables don't have such searches, so it makes sense for alignment. I just wanted to mention it.)

Reply:
@hackrm thoughts?

Reply:
No need to include searching by year, don't see a lot of value with it. Keeping it consistent with the other tables is good.